### DISCUSSION TITLE
Skip fetching downloads in tycho-p2-repository-plugin:assemble-repository step for <require> features

### DISCUSSION THREAD

**Discussion by valentinene** [eclipse-tycho/tycho#1844] [Q&A] [answered]
I have an eclipse repository which includes a feature on which I added a requirement for another feature in its feature.xml in <requires> <import> tags, I am using tycho 2.2.0, and this required feature includes and requires a LOT of other features and plugins, basically another eclipse IDE. I added the nexus repository in which this required feature is found, but the build is taking up a lot of time.
One thing would by my nexus repository being slow downloading almost 1GB of dependencies, but since these requirements are only required by the existing IDE, and not included in my eclipse repository why does the build HAVE to download them and could I force it to not need to resolve it at build time?
Is there a way we can skip downloading the artifacts for features which are from the <requires> tag in my feature.xml  when we assemble the eclipse repository which contains it?
Since these artifacts are not included in the repository is there a way to maybe skip the p2 dependency resolution and assembly part for them?

**Comment by laeubi** [comment, marked as answer]
Tycho 2.2.0 is outdated, please Upgrade to Tycho 3.0.1

Beside that it is hard to guess *why* anything is downloaded without an example as it depends on your configuration as well. Anyways Tycho should cache downloaded data in your local repository.

**Reply by valentinene** [reply]
In my project I have some bundles, and one feature which includes those bundles as well as some other features. I went and added a requirement to a certain version of a feature which is found in our product, basically to force using my project only for versions higher than specified: 
`<requires> 
		<import feature="com.my.required.feature" version="1.1.0" match="greaterOrEqual"/> 
</requires>`
This feature from the <requires> comes from a p2 repository which is downloaded from an internal nexus repository in a <repository> tag in my parent pom.
Before the actual build starts and the p2 repositories are added the time delay is manageable. But when the build reaches tycho-p2-repository-plugin:2.2.0:assemble-repository (default-assemble-repository) @ com.my.repository and it starts fetching all the artifacts from my required feature's p2 repository it takes a lot of time, there are a lot of them, basically this feature contains a ton of plugins and other features. This part I am looking to somehow skip, since these artifacts are not used at all, only one feature is added as a requirement in my feature but for the platform my repository will be installed into, could the build somehow skip all this downloading and just trust the platform will solve this requirement?

**Reply by laeubi** [reply]
Tycho only download things if they are required, so you should check your configuration there. for example https://tycho.eclipseprojects.io/doc/latest/tycho-p2/tycho-p2-repository-plugin/assemble-repository-mojo.html#includeAllDependencies

**Reply by valentinene** [reply]
Hm, more interesting I find the `<includeRequiredFeatures>` option, but I did not see these options in my older tycho version. I`ll update to 3.0.1 and see if it helps my issue.

**Reply by valentinene** [reply]
yeah, tycho 3.0.1 and parallel builds did the trick insanely, thanks for bringin me up to the 20th century :D
